packages/kbn-language-documentation-popover/src/components/as_flyout/index.test.tsx

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import React from 'react';
+import { screen, render, fireEvent, waitFor } from '@testing-library/react';
+import { LanguageDocumentationFlyout } from '.';
+
+jest.mock('../../sections', () => {
+  const module = jest.requireActual('../../sections');
+  return {
+    ...module,
+    getESQLDocsSections: () => ({
+      groups: [
+        {
+          label: 'Section one',
+          description: 'Section 1 description',
+          items: [],
+        },
+        {
+          label: 'Section two',
+          items: [
+            {
+              label: 'Section two item 1',
+              description: 'Section two item 1 description',
+            },
+            {
+              label: 'Section two item 2',
+              description: 'Section two item 2 description',
+            },
+          ],
+        },
+        {
+          label: 'Section three',
+          items: [
+            {
+              label: 'Section three item 1',
+              description: 'Section three item 1 description',
+            },
+            {
+              label: 'Section three item 2',
+              description: 'Section three item 2 description',
+            },
+          ],
+        },
+      ],
+      initialSection: <span>Here is the initial section</span>,
+    }),
+  };
+});
+
+describe('###Documentation flyout component', () => {
+  const renderFlyout = (linkToDocumentation?: string) => {
+    return render(
+      <LanguageDocumentationFlyout
+        isHelpMenuOpen={true}
+        onHelpMenuVisibilityChange={jest.fn()}
+        linkToDocumentation={linkToDocumentation}
+      />
+    );
+  };
+  it('has a header element for navigation through the sections', () => {
+    renderFlyout();
+    expect(screen.getByTestId('language-documentation-navigation-search')).toBeInTheDocument();
+    expect(screen.getByTestId('language-documentation-navigation-dropdown')).toBeInTheDocument();
+    expect(screen.queryByTestId('language-documentation-navigation-link')).not.toBeInTheDocument();
+  });
+
+  it('has a link if linkToDocumentation prop is given', () => {
+    renderFlyout('meow');
+    expect(screen.getByTestId('language-documentation-navigation-link')).toBeInTheDocument();
```

Comment:
nit: this assertion is redundant. If the test ID isn't in the document, `screen.getByTestId` will throw an error before you ever get to the `expect().toBeInTheDocument`. 

`screen.queryByTestId`, on the other hand, will not throw an error and will return `null`.

Functionally, I guess it's pretty much the same.